packages/material-ui/src/Box/Box.d.ts

```diff
@@ -34,14 +451,8 @@ export type BoxStyleFunction = ComposedStyleFunction<
 
 type SystemProps = PropsFor<BoxStyleFunction>;
 type ElementProps = Omit<React.HTMLAttributes<HTMLElement>, keyof SystemProps>;
-type SxPropsValue = Omit<CSSObject, keyof SystemProps> & SystemProps;
 
-export type SxProps = {
-  [Name in keyof SxPropsValue]?:
-    | SxPropsValue[Name]
-    | ((theme: Theme) => CSSObject | SxPropsValue[Name])
-    | SxProps;
-};
+export type SxProps = SystemStyleObject;
```

Comment:
I wonder if we shouldn't expose this type from `@material-ui/system`.

Reply:
I think it makes sense yes, we just need to parameterize the Theme so that in the core it can reflect our theme structure. Let me try to move the typings to see if there will be any issues...

Reply:
Exported from there and added generic for the `Theme` so that in the core components we can have the Theme structure from `createMuiTheme`. Checked, worked correctly locally, let's see if we will have a green build. 👍